includes/class-wc-payments-db.php

```diff
@@ -30,6 +30,38 @@ public function order_from_charge_id( $charge_id ) {
 		return false;
 	}
 
+	/**
+	 * Retrieve orders from the DB using a list of corresponding Stripe charge IDs.
+	 *
+	 * @param array $charge_ids List of charge IDs corresponding to an order ID.
+	 *
+	 * @return boolean|WC_Order|WC_Order_Refund
+	 */
+	public function orders_with_charge_id_from_charge_ids( $charge_ids ) {
+		global $wpdb;
+
+		$charge_id_placeholder = join( ',', array_fill( 0, count( $charge_ids ), '%s' ) );
+
+		// The order ID is saved to DB in `WC_Payment_Gateway_WCPay::process_payment()`.
+		$results = $wpdb->get_results(
+			$wpdb->prepare(
+				// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared,WordPress.DB.PreparedSQLPlaceholders.UnfinishedPrepare
+				"SELECT DISTINCT ID as order_id, meta.meta_value as charge_id FROM $wpdb->posts as posts LEFT JOIN $wpdb->postmeta as meta ON posts.ID = meta.post_id WHERE meta.meta_key = '_charge_id' AND meta.meta_value IN ($charge_id_placeholder)",
+				$charge_ids
+			)
+		);
+
+		return array_map(
+			function ( $row ) {
```

Comment:
Which type `$row` has? We also can declare `: array` as the closure return type.